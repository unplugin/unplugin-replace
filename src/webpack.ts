/**
 * This entry file is for webpack plugin.
 *
 * @module
 */

import unplugin from './index'

/**
 * Webpack plugin
 *
 * @example
 * ```js
 * // webpack.config.js
 * import Replace from 'unplugin-replace/webpack'
 *
 * default export {
 *  plugins: [Replace()],
 * }
 * ```
 */
const webpack = unplugin.webpack as typeof unplugin.webpack
export default webpack
