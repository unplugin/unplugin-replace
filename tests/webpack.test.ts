import { mkdtemp, readFile, rm } from 'node:fs/promises'
import os from 'node:os'
import path from 'node:path'
import { expect, onTestFinished, test } from 'vitest'
import webpack from 'webpack'
import UnpluginReplace from '../src/webpack'

test('webpack', async () => {
  const tmp = await mkdtemp(path.join(os.tmpdir(), 'unplugin-replace-'))
  onTestFinished(() => rm(tmp, { recursive: true, force: true }))

  await new Promise<webpack.Stats>((resolve, reject) => {
    const compiler = webpack({
      entry: path.resolve(__dirname, 'fixtures/main.js'),
      output: {
        path: tmp,
      },
      plugins: [
        UnpluginReplace({
          'process.platform': '"darwin"',
        }),
      ],
      mode: 'production',
      target: 'node',
      optimization: {
        minimize: false,
      },
    })
    compiler!.run((error, stats) => {
      if (error) return reject(error)
      resolve(stats!)
    })
  })

  const result = await readFile(path.resolve(tmp, 'main.js'), 'utf8')
  expect(result).toMatchSnapshot()
  expect(result).not.contains('process.platform')
  expect(result).contains('"darwin"')
})
