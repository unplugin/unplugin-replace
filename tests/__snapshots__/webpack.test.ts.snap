// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`webpack 1`] = `
"/******/ (() => { // webpackBootstrap
/******/ 	"use strict";

// UNUSED EXPORTS: platform

;// external "node:process"
const external_node_process_namespaceObject = require("node:process");
;// ./tests/fixtures/main.js


const platform = "darwin"

console.log('hello "darwin"', platform)
console.log(DEV)

/******/ })()
;"
`;
